### PR TITLE
ASC-173-Integrate with Molecule validation APIs

### PR DESCRIPTION
**Pull request by Sneha-SR**
ASC-173:
1. Molecule Validation API,
2. Molecule upload API.
3. Upload molecule file API.
4. Download CSV file for rejected smiles.
5. Unit test cases
 
Need to add new key API_END_POINT and URL for APIs in .env file under web

1.Integrated ASC-580 Save molecule to ketcher tool
2. IntegratedASC-586 : Import molecules from file
3. Implemented Download CSV template in add molecule

### CODE REVIEW

**Comment by Sneha-SR** [comment]
[CSV_Sample_Data.csv](https://github.com/user-attachments/files/17557840/CSV_Sample_Data.csv)
@himanshubexcelra this the sample file to upload smiles

File: apps/web/components/Molecule/service.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import {
+    UploadMoleculeFileRequest,
+    UploadMoleculeSmilesRequest,
+    ValidateSmileRequest
+} from "../../lib/definition";
+import CustomFile from "../../utils/file";
+
+// const sampleMolecules = {
+//     "smiles": "Cc1ccccc1"
```
Comment (himanshubexcelra) [inline apps/web/components/Molecule/service.ts:9]: remove comments.

File: apps/web/components/Molecule/service.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            const error = await response.json();
+            return { status: response.status, error };
+        }
+    } catch (error: any) {
+        return error;
+    }
+}
+
+// const sampleMolecules = {
+//     "smiles": ["Cc1ccccc1", "C1=CCC=CC1"],
```
Comment (himanshubexcelra) [inline apps/web/components/Molecule/service.ts:38]: Remove comments

File: apps/web/components/Molecule/service.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        formData.append('organization_id', data.organization_id);
+        formData.append('updated_by_user_id', data.updated_by_user_id);
+        const response = await fetch(`${process.env.API_END_POINT}/upload_molecule_files`, {
+            method: "POST",
+            body: formData,
+        });
+        if (response.status === 200) {
+            const data = await response.json();
+            // Can use following code to download rejected Smiles
+            // const header = {smiles: "SMILE", reason: "Reason"}
```
Comment (himanshubexcelra) [inline apps/web/components/Molecule/service.ts:84]: Remove comment

**Pull request merged**